Dockerfile

```diff
@@ -33,13 +38,33 @@ RUN apt-get update && apt-get install -y \
     libgflags-dev \
     libgoogle-glog-dev
 
-WORKDIR /sim
+ARG rocm_ver=1.6.2
 
-ARG rocm_ver=1.6.0
+# Get files needed for gem5, and apply patches
+RUN git clone --single-branch --branch agutierr/master-gcn3-staging https://gem5.googlesource.com/amd/gem5
 
+RUN git clone --single-branch https://github.com/ROCm-Developer-Tools/HIP/
+RUN git clone --single-branch https://github.com/ROCmSoftwarePlatform/hipBLAS/
+RUN git clone --single-branch https://github.com/ROCmSoftwarePlatform/rocBLAS/
+RUN git clone --single-branch https://github.com/ROCmSoftwarePlatform/MIOpenGEMM/
+RUN git clone --single-branch https://github.com/ROCmSoftwarePlatform/MIOpen/
+RUN git clone --single-branch https://github.com/RadeonOpenCompute/rocm-cmake/
+
+
+# Get and apply patches to various repos
+COPY patch /patch
+
+RUN git -C /gem5/ apply /patch/gem5.patch
+RUN git -C /HIP/ checkout 0e3d824e && git -C /HIP/ apply /patch/hip.patch
+RUN git -C /hipBLAS/ checkout ee57787e && git -C /hipBLAS/ apply /patch/hipBLAS.patch
+RUN git -C /rocBLAS/ checkout cbff4b4e && git -C /rocBLAS/ apply /patch/rocBLAS.patch
+RUN git -C /MIOpenGEMM/ checkout 9547fb9e
+RUN git -C /MIOpen/ checkout a9949e30 && git -C /MIOpen/ apply /patch/miopen.patch
```

Comment:
this may need to become more complex if we are adding multiple patches in the future.  Potentially we could just create our own fork of MIOpen, put those patches on, and check that out?  Fine for now though.

Reply:
Creating a separate branch to checkout would work, it's what I do when cloning gem5.